### PR TITLE
Pin dep version to 4.1.

### PR DESCRIPTION
**Pull request by echlebek**
Signed-off-by: Eric Chlebek <eric@sensu.io>

## What is this change?

It makes sure we are using a specific dep version to build our PRs.

## Why is this change necessary?

Previously we were using master of dep.

## Does your change need a Changelog entry?

NEIN

## Do you need clarification on anything?

NEIN

## Were there any complications while making this change?

There probably will be tbh

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -44,12 +44,6 @@ install_deps () {
     go get github.com/jgautheron/goconst/cmd/goconst
     go get honnef.co/go/tools/cmd/megacheck
     go get github.com/golang/lint/golint
-    install_golang_dep
-}
-
-install_golang_dep() {
-    go get github.com/golang/dep/cmd/dep
```
Comment (nikkiattea) [inline build.sh]: What if a user (not CI) does not have dep installed? They'll just get an error?

Comment (echlebek) [reply build.sh]: That's right. They will have to install dep on their own.

Comment (echlebek) [reply build.sh]: Yeah, we should add it to the getting started instructions.

The problem is that we can't install dep for people without making assumptions or adding complexity. Similarly to how we can't install Go for people.

**Comment by echlebek** [comment]
I added some notes about Go and dep to the readme.

**Pull request merged**